iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/load/service/TopologyService.java

```diff
@@ -228,7 +228,7 @@ private synchronized void topologyProbing() {
           && !failureDetector.isAvailable(entry.getKey(), entry.getValue())) {
         LOGGER.debug("Connection from DataNode {} to DataNode {} is broken", fromId, toId);
       } else {
-        latestTopology.get(fromId).add(toId);
+        Optional.ofNullable(latestTopology.get(fromId)).ifPresent(s -> s.add(toId));
```

Comment:
Why not just computeIfAbsent? Intuitively, I bet the bug is merely due to ur `latestTopology` lost some keys. And the `latestTopology` should cache all nodeIds as the key

Reply:
Let me change to computeIfAbsent. Some keys are **intentionally** left out because these datanodes are starting and cannot answer to probing RPC requests.